trin-core/src/portalnet/overlay.rs

```diff
@@ -138,9 +143,111 @@ impl OverlayProtocol {
             data_radius: Arc::new(RwLock::new(data_radius)),
             kbuckets,
             db,
+            utp_listener,
         }
     }
 
+    pub async fn process_response(
+        &self,
+        response_vec: Vec<u8>,
+        enr: Enr,
+        content_keys_vec: Vec<Vec<u8>>,
+    ) -> Result<(), String> {
```

Comment:
We ought to return a non-`String` error type here. Instead, we can use `OverlayRequestError`.